tests/test_operation.py

```diff
@@ -709,6 +709,23 @@ def test_operation_multiply_invalid(self):
             T = X @ Z
             Y @ T
 
+    def test_eigvals(self):
+        """Test that the correct eigenvalues are returned for the Tensor"""
+        X = qml.PauliX(0)
+        Y = qml.PauliY(2)
+        t = Tensor(X, Y)
+        assert np.array_equal(t.eigvals, np.kron(qml.PauliX.eigvals, qml.PauliY.eigvals))
+
+    def test_eigvals_hermitian(self, tol):
+        """Test that the correct eigenvalues are returned for the Tensor"""
+        X = qml.PauliX(0)
+        hamiltonian = np.array([[1,0,0,0], [0,1,0,0], [0,0,0,1], [0,0,1,0]])
+        Herm = qml.Hermitian(hamiltonian, wires=[1, 2])
+        t = Tensor(X, Herm)
+        d = np.kron(np.array([1., -1.]), np.array([-1.,  1.,  1.,  1.]))
+        t = t.eigvals
+        assert np.allclose(t, d, atol=tol, rtol = 0)
```

Comment:
It would be great to also include a test for a tensor observable that includes a mix of I, paulis, and Hermitians, since I/Hermitian are the two special cases.

Reply:
As per our discussion, ``Identity`` might not be compatible with qubit operations, as it is defined for CV now inheriting from ``CVObservable`` (noting it just so that there is a follow-up on this here as well).